packages/ckeditor5-undo/src/basecommand.js

```diff
@@ -93,15 +93,25 @@ export default class BaseCommand extends Command {
 		const selectionRanges = [];
 
 		// Transform all ranges from the restored selection.
-		for ( const range of ranges ) {
-			const transformed = transformSelectionRange( range, operations );
+		const transformedRangeGroups = ranges.map( range => range.getTransformedByOperations( operations ) );
+		const allRanges = transformedRangeGroups.flat();
+
+		for ( const rangeGroup of transformedRangeGroups ) {
+			// While transforming there could appear ranges that are contained by other ranges, we shall ignore them.
+			const transformed = rangeGroup.filter( range => !allRanges.some( otherRange => (
```

Comment:
I don't like this part of code, because having `.fliter()` then inside `.some()`, which operate on similar but different data, it seems clunky and reads bad. Can we do that in a different way?

Reply:
Or have the inside function more clear?